src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,210 @@
+import randoop.util.Randomness;
+import randoop.util.SimpleArrayList;
+
+/**
+ * Implements the Bloodhound component, as described by the Guided Random Testing (GRT) paper.
+ * Bloodhound computes a weight for each method under test by taking a weighted combination of the
+ * uncovered branch ratio and the ratio between the number of times the method has been invoked and
+ * the maximum number of times any method under test has been invoked.
+ *
+ * <p>However, some hyper-parameters and edge cases were left unspecified in the GRT paper. We have
+ * chosen our own values for the following unspecified hyper-parameters:
+ *
+ * <ul>
+ *   <li>alpha - parameter to balance branch coverage and number of invocations when computing
+ *       weight.
+ *   <li>p - parameter for decreasing weights of methods between updates to coverage information.
+ *   <li>Interval for recomputing branch coverage information.
+ *   <li>Default weight for cases where a method has zero branches or computed weight is zero.
+ * </ul>
+ */
+public class Bloodhound implements TypedOperationSelector {
+
+  /**
+   * Map of methods under test to their weights. These weights are dynamic and depend on branch
+   * coverage.
+   */
+  private final Map<TypedOperation, Double> methodWeights = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have been recently selected by the {@link
+   * ForwardGenerator} to construct a new sequence. This map is cleared every time branch coverage
+   * is recomputed.
+   */
+  private final Map<TypedOperation, Integer> methodSelections = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have ever been selected by the {@link
+   * ForwardGenerator} to extend an existing sequence to construct a new sequence. This definition
+   * is the same as that of {@code methodSelections} except that we do not clear this map every time
+   * we recompute branch coverage. Thus, the integer value for a given method is non-decreasing
+   * during a run of Randoop. The GRT paper does not state its definition of the "number of
+   * invocations" of a method under test.
+   */
+  private final Map<TypedOperation, Integer> methodSuccessfulCalls = new HashMap<>();
+
+  /** List of operations, identical to ForwardGenerator's operation list. */
+  private final SimpleArrayList<TypedOperation> operationSimpleList;
+
+  /**
+   * Hyper-parameter for balancing branch coverage and number of times a method was chosen. The name
+   * alpha is from the GRT paper.
+   */
+  private final double alpha = 0.7;
+
+  /**
+   * Hyper-parameter for decreasing weights of methods between updates to coverage information. The
+   * name p is from the GRT paper.
+   */
+  private final double p = 0.5;
+
+  /** How often to recompute branch coverage, in steps. */
+  private final int branchCoverageInterval = 100;
+
+  /** Maximum number of successful calls so far to any method under test. */
```

Comment:
What is the definition of "successful calls" as opposed to "calls" or "selections"?

Is the word "successful" motivated by "succ(m)" in the GRT paper?  Please connect the dots for me.

Reply:
Right, I originally used "successful" due to the GRT paper's definition of succ(m) as the number of "successful invokations" or a method. However, we concluded that the GRT paper didn't define how they defined "successful invokation", we decided to use the number of times a method is selected to construct a new and unique sequence for the definition of `succ(m)` and `maxSucc(M)`. I've rewritten usages of "successful calls/invokations" to "selections".